mne/chpi.py

```diff
@@ -478,10 +478,13 @@ def _fit_cHPI_amplitudes(raw, time_sl, hpi, fit_time, verbose=None):
 
     data_diff = np.dot(model, X).T - this_data
 
-    # compute amplitude correlation (For logging)
-    g_sin = 1 - np.sqrt((data_diff**2).sum() / (this_data**2).sum())
-    g_chan = 1 - np.sqrt((data_diff**2).sum(axis=1) /
-                         (this_data**2).sum(axis=1))
+    # compute amplitude correlation (for logging), protect against zero
+    norm = (this_data ** 2).sum(axis=1)
```

Comment:
if this_data is not too small I suspect

norm = np.linalg.norm(this_data, axis=1)**2

avoid the allocation of this_data ** 2